### PR TITLE
feat: support enabling web via a flag in dev mode

### PR DESCRIPTION
**Pull request by mvayngrib**
## Description

this lets us turn on web mode when using JSC by setting this env var

### CODE REVIEW

**Review by gutenye — APPROVED** [review]
utacK

File: src/reanimated2/PlatformChecker.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export function isWeb(): boolean {
 }
 
 export function shouldBeUseWeb() {
-  return isJest() || isChromeDebugger() || isWeb();
+  return isJest() || isChromeDebugger() || isWeb() || ((global as any).__DEV__ && process.env.REANIMATED_WEB === 'true');
```
Comment (alexandrius) [inline src/reanimated2/PlatformChecker.ts:16]: I wonder why `isChromeDebugger()` fails?

Comment (alexandrius) [reply src/reanimated2/PlatformChecker.ts:16]: We should fix it instead

Comment (mvayngrib) [reply src/reanimated2/PlatformChecker.ts:16]: agree, just not sure how to detect

Comment (alexandrius) [reply src/reanimated2/PlatformChecker.ts:16]: @mvayngrib maybe let's add new function and extract it there? Or check how `isChromeDebugger()` is implemented in Reanimated 3?

**Comment by mvayngrib** [comment]
might close this if this pans out https://exodusio.slack.com/archives/CFKF4FDTJ/p1720197467825229

if we can get it working with a JS change only, that seems simplest
```
const {
  default: NativeDevSettings,
} = require('react-native/Libraries/NativeModules/specs/NativeDevSettings')
NativeDevSettings.setIsDebuggingRemotely(true)
```

**Comment by mvayngrib** [comment]
not needed after https://github.com/ExodusMovement/exodus-mobile/pull/20595

**Pull request closed without merging**